### ISSUE TITLE
Menu - Submenu remains visible even after closing the parent menu

### ISSUE THREAD

**Issue by SpaNb4** [ant-design/ant-design#43267]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/antd-reproduction-template-forked-mlq88x?file=/index.js)

### Steps to reproduce

1. Click on the "Click me" button to open the dropdown menu
2. Click on the "Submenu" item
3. Press the Esc key to close the parent menu

### What is expected?

Submenu is closed

### What is actually happening?

Submenu remains visible

| Environment | Info |
| --- | --- |
| antd | 5.6.3 |
| React | 18.2.0 |
| System | Pop!_OS 22.04 LTS |
| Browser | Version 114.0.5735.198 |

**Comment by MadCcc** [comment]
Should focue submenu when opened.

**Comment by heiyu4585** [comment]
结论: Menu 自己维护的 active，这个有点难改的

---

1.   subMenu子菜单hover的时候触发一次 [onAcitve](https://github.com/react-component/menu/blob/d896516bf648a7c0eec71c7315a5646eaf110d22/src/hooks/useActive.ts#L35),当前的aciveKey 就是 当前的hover子菜单 1-2
2.   键盘事件->useAccessibility -> 会触发到[offsetObj.*offset* < 0](https://github.com/react-component/menu/blob/d896516bf648a7c0eec71c7315a5646eaf110d22/src/hooks/useAccessibility.ts#L358)  触发一次关闭父级菜单的操作 [triggerAccessibilityOpen](https://github.com/react-component/menu/blob/d896516bf648a7c0eec71c7315a5646eaf110d22/src/hooks/useAccessibility.ts#L365)]

**Comment by tomislavdimitrov2003** [comment]
Issues is still present. Will there be a fix for it soon?

**Comment by yoyo837** [comment]
https://github.com/react-component/menu/pull/671 Does this fix?

**Comment by SpaNb4** [comment]
> [react-component/menu#671](https://github.com/react-component/menu/pull/671) Does this fix?

No, I can still reproduce the issue with `rc-menu` 9.12.3 installed: https://codesandbox.io/p/sandbox/antd-reproduction-template-forked-4rzmn8

**Comment by yoyo837** [comment]
https://codesandbox.io/p/sandbox/antd-reproduction-template-forked-vyqqlh?file=%2Findex.js%3A35%2C16

Seems to work fine without using the deprecated `overlay` usage.

**Comment by yoyo837** [comment]
It looks like the issue is related to `triggerSubMenuAction="click"`.

**Comment by noxxxxxxxx** [comment]
Still exist in the latest version
![image](https://github.com/ant-design/ant-design/assets/37919656/2a1ccf3a-b1c8-44f5-a6c3-48cdf8aec9a9)
[codesandbox](https://codesandbox.io/p/sandbox/antd-reproduction-template-forked-cpffsp?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clvl7em2e00072e689yg59zpa%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clvl7em2e00032e685m7ekluv%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clvl7em2e00042e68nb1ewr3g%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clvl7em2e00062e682m2eokwk%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clvl7em2e00032e685m7ekluv%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvl7em2e00022e683w5fnxtq%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A10%252C%2522startColumn%2522%253A28%252C%2522endLineNumber%2522%253A10%252C%2522endColumn%2522%253A28%257D%255D%252C%2522filepath%2522%253A%2522%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clvl7em2e00032e685m7ekluv%2522%252C%2522activeTabId%2522%253A%2522clvl7em2e00022e683w5fnxtq%2522%257D%252C%2522clvl7em2e00062e682m2eokwk%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvl7em2e00052e68vree1yck%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clvl7em2e00062e682m2eokwk%2522%252C%2522activeTabId%2522%253A%2522clvl7em2e00052e68vree1yck%2522%257D%252C%2522clvl7em2e00042e68nb1ewr3g%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clvl7em2e00042e68nb1ewr3g%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by EmilyyyLiu** [comment]
Reproduction result
  
  I followed the reproduction steps from this issue and can no longer reproduce the originally described behavior 
  ("submenu remains visible"):

  - Open the dropdown → open the submenu → press Esc, both the dropdown and the submenu popup close correctly.

  A related behavior still observed (would like to confirm if expected)

  After opening both the dropdown and the submenu, when focus stays on the parent popup (the dropdown menu itself)
  and Esc is pressed:
  - the parent popup closes;
  - but the submenu popup does not close along with it and remains visible on screen.

  When focus is inside the submenu (on a submenu item) and Esc is pressed, both layers close as expected.

  Could you confirm whether this behavior — "focus on parent + Esc does not close the submenu, leaving it visible" —
  is expected, or whether it should also be fixed?

**Comment by SpaNb4** [comment]
> Reproduction result
> 
> I followed the reproduction steps from this issue and can no longer reproduce the originally described behavior ("submenu remains visible"):
> 
> * Open the dropdown → open the submenu → press Esc, both the dropdown and the submenu popup close correctly.
> 
> A related behavior still observed (would like to confirm if expected)
> 
> After opening both the dropdown and the submenu, when focus stays on the parent popup (the dropdown menu itself) and Esc is pressed:
> 
> * the parent popup closes;
> * but the submenu popup does not close along with it and remains visible on screen.
> 
> When focus is inside the submenu (on a submenu item) and Esc is pressed, both layers close as expected.
> 
> Could you confirm whether this behavior — "focus on parent + Esc does not close the submenu, leaving it visible" — is expected, or whether it should also be fixed?

Thanks for revalidating! Can confirm the original issue no longer reproduces on the latest antd version.

However, I found another related issue: open the dropdown menu → open the submenu → press Tab twice (or once if focus is on the submenu). The parent popup closes but the submenu stays visible. Video attached: [video](https://github.com/user-attachments/assets/a8a3dd0f-01b0-4853-9cfe-e54130a704b2)
Sandbox to reproduce: https://codesandbox.io/p/devbox/antd-reproduction-template-forked-8hdkf3

I also tried to reproduce the issue you described, but couldn't figure out how to keep focus on the parent popup while the submenu is open. Either way, that behavior does seem like a bug too.
